java-security/src/test/java/com/sap/cloud/security/servlet/XsuaaTokenAuthenticatorTest.java

```diff
@@ -105,13 +104,29 @@ public void validateIasToken_WhenRestClientIsNull() throws Exception {
 				.execute(() -> {
 					cut = new XsuaaTokenAuthenticator()
 							.withHttpClient(null)
-							.withServiceConfiguration(oAuth2ServiceConfiguration);
+							.withServiceConfiguration(oAuth2ServiceConfigBuilder.build());
 					response[0] = cut.validateRequest(httpRequest, HTTP_RESPONSE);
 				});
 		assertEquals("IAS token validation is not supported: setup is malicious.",
 				response[0].getUnauthenticatedReason());
 	}
 
+	@Test
+	public void validateUaaToken() {
+		HttpServletRequest httpRequest = createRequestWithToken(uaaToken.getTokenValue());
+
+		cut = new XsuaaTokenAuthenticator()
+				.withHttpClient(mockHttpClient)
+				.withServiceConfiguration(oAuth2ServiceConfigBuilder.withClientId("dashboard_client-Id").build());
```

Comment:
should there also be a issuer?

Reply:
Git it. That is in "@Before"

Reply:
oAuth2ServiceConfig/VCAP_Services is setup here:
https://github.com/SAP/cloud-security-xsuaa-integration/blob/6bb1ef3342a040a6e62a8cc6ebe09de6fef1f6a2/java-security/src/test/java/com/sap/cloud/security/servlet/XsuaaTokenAuthenticatorTest.java#L74-L80
the token of course contains an issuer, which is 
```
"iss": "http://localhost:62700",
```
which is different to what we have in the service configuration, but in XSUAA the "Issuer-Validation" is performed on jku header (instead of iss claim) as issuer is not used.

Reply:
see also here:
https://github.com/SAP/cloud-security-xsuaa-integration/blob/main/java-security/src/main/java/com/sap/cloud/security/token/validation/validators/XsuaaJkuValidator.java